sdk/core/core-amqp/src/auth/sas.ts

```diff
@@ -78,7 +78,46 @@ export class SharedKeyCredential {
    * @param {string} connectionString - The EventHub/ServiceBus connection string
    */
   static fromConnectionString(connectionString: string): SharedKeyCredential {
-    const parsed = parseConnectionString<ServiceBusConnectionStringModel>(connectionString);
-    return new SharedKeyCredential(parsed.SharedAccessKeyName, parsed.SharedAccessKey);
+    const parsed = parseConnectionString<
+      ServiceBusConnectionStringModel & { SharedAccessSignature: string }
+    >(connectionString);
+
+    if (parsed.SharedAccessSignature == null) {
+      return new SharedKeyCredential(parsed.SharedAccessKeyName, parsed.SharedAccessKey);
+    } else {
+      return new SharedAccessSignatureCredential(parsed.SharedAccessSignature);
+    }
+  }
+}
+
+/**
+ * A credential that takes a SharedAccessSignature:
+ * `SharedAccessSignature sr=<resource>&sig=<signature>&se=<expiry>&skn=<keyname>`
+ *
+ * @internal
+ * @ignore
+ */
+export class SharedAccessSignatureCredential extends SharedKeyCredential {
+  private _accessToken: AccessToken;
+
+  /**
+   * @param sharedAccessSignature A shared access signature of the form `sr=<resource>&sig=<signature>&se=<expiry>&skn=<keyname>`
```

Comment:
shouldnt this have the SAS prefix?

```suggestion
   * @param sharedAccessSignature A shared access signature of the form `SharedAccessSignature sr=<resource>&sig=<signature>&se=<expiry>&skn=<keyname>`
```

Reply:
Yes. Fixed.